iroh/src/commands/sync.rs

```diff
@@ -0,0 +1,263 @@
+};
+
+use super::RpcClient;
+
+const MAX_DISPLAY_CONTENT_LEN: u64 = 1024 * 1024;
+
+#[allow(clippy::large_enum_variant)]
+#[derive(Debug, Clone, Parser)]
+pub enum Commands {
+    Author {
+        #[clap(subcommand)]
+        command: Author,
+    },
+    Docs {
+        #[clap(subcommand)]
+        command: Docs,
+    },
+    Doc {
+        id: NamespaceId,
+        #[clap(subcommand)]
+        command: Doc,
+    },
+}
+
+impl Commands {
+    pub async fn run(self, client: RpcClient) -> anyhow::Result<()> {
+        let iroh = Iroh::new(client);
+        match self {
+            Commands::Author { command } => command.run(iroh).await,
+            Commands::Docs { command } => command.run(iroh).await,
+            Commands::Doc { command, id } => command.run(iroh, id).await,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Parser)]
+pub enum Author {
+    List,
+    Create,
+}
+
+impl Author {
+    pub async fn run(self, iroh: Iroh) -> anyhow::Result<()> {
+        match self {
+            Author::List => {
+                let mut stream = iroh.list_authors().await?;
+                while let Some(author_id) = stream.try_next().await? {
+                    println!("{}", author_id);
+                }
+            }
+            Author::Create => {
+                let author_id = iroh.create_author().await?;
+                println!("{}", author_id);
+            }
+        }
+        Ok(())
+    }
+}
+
+#[derive(Debug, Clone, Parser)]
+pub enum Docs {
+    List,
+    Create,
+    // Import {
+    //     key: String,
+    //     #[clap(short, long)]
+    //     peers: Vec<PeerSource>,
+    // },
+    Import { ticket: DocTicket },
+}
+
+impl Docs {
+    pub async fn run(self, iroh: Iroh) -> anyhow::Result<()> {
+        match self {
+            Docs::Create => {
+                let doc = iroh.create_doc().await?;
+                println!("created {}", doc.id());
+            }
+            // Docs::Import { key, peers } => {
+            //     let key = hex::decode(key)?
+            //         .try_into()
+            //         .map_err(|_| anyhow!("invalid length"))?;
+            //     let ticket = DocTicket::new(key, peers);
+            //     let doc = iroh.import_doc(ticket).await?;
+            //     println!("imported {}", doc.id());
+            // }
+            Docs::Import { ticket } => {
+                let doc = iroh.import_doc(ticket).await?;
+                println!("imported {}", doc.id());
+            }
+            Docs::List => {
+                let mut stream = iroh.list_docs().await?;
+                while let Some(id) = stream.try_next().await? {
+                    println!("{}", id)
+                }
+            }
+        }
+        Ok(())
+    }
+}
+
+#[derive(Debug, Clone, Parser)]
+pub enum Doc {
+    StartSync {
+        peers: Vec<PeerSource>,
+    },
+    Share {
+        mode: ShareMode,
+    },
+    /// Set an entry
+    Set {
+        /// Author of this entry.
+        author: AuthorId,
```

Comment:
should be the same as with get `#[clap(short, long)]` otherwise this is a positional argument

Reply:
Changed. However also this is all overhauled in #1356